src/main/java/org/apache/commons/beanutils2/converters/DurationConverter.java

```diff
@@ -74,10 +84,21 @@ public DurationConverter(final Object defaultValue) {
     @Override
     protected <T> T convertToType(final Class<T> type, final Object value) throws Throwable {
         if (Duration.class.equals(type)) {
-            return type.cast(Duration.parse((String.valueOf(value))));
+            final String stringValue = toString(value);
+
+            try {
+                return type.cast(Duration.parse(stringValue));
+            } catch (DateTimeParseException ex) {
+                try {
+                    long number = Long.parseLong(stringValue);
+                    return type.cast(Duration.of(number, ChronoUnit.MILLIS));
```

Comment:
**-1**: This feels like a recipe for problems and confusion, the code expects _some_ number in _some_ format in _some_ scale. This is just not a Duration which is a well defined concept.